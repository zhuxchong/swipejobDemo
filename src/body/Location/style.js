import styled from "styled-components";

export const LocationWrapper = styled.div`
  border-top: solid gray 1px;
  border-bottom: solid gray 1px;
  width: 100%;
  padding: 3% 0%;
  display: flex;
  flex-direction: column;
  flex: 1.5;
  h4 {
    margin-top: 0;
    margin-bottom: 10px;
    text-align: center;
    line-height: 20px;
  }
  p {
    margin: 0;
    line-height: 20px;
    text-align: center;
    color: #0066cc;
    text-decoration: underline;
  }
  @media (max-width: 375px) {
    padding: 2% 0%;
    h4 {
      font-size: 15px;
      line-height: 12px;
    }
    p {
      font-size: 10px;
      line-height: 12px;
    }
  }
`;
