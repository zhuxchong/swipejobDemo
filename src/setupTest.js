import { configure } from "enzyme";
import Ada from "enzyme-adapter-react-16";
//set up
configure({ adapter: new Ada() });
