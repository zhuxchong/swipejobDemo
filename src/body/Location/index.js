import React from "react";
import { LocationWrapper } from "./style";
const Location = props => {
  const { location } = props;
  return (
    <LocationWrapper>
      <h4>Location</h4>
      <p className="detailLocation">{location}</p>
    </LocationWrapper>
    //<div>123{console.log(props)}</div>
  );
};
export default Location;
