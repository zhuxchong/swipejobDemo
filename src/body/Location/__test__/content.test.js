import React from "react";
import { shallow } from "enzyme";
import Location from "../index";
require("../../../setupTest");
const props = {
  //   title: "Construction and Releated Workers - General",
  //   companyName: "C.D. Barnes and Associates",
  //   wagePerHour: "$8.38/hour",
  //   workPeriod: "Mon, Sep 05 - Fri, Sep 16",
  location: "123 Main Street, Tacoma, WA 98409"
};
let wrapper;
describe("Description component", () => {
  beforeEach(() => {
    wrapper = shallow(<Location {...props} />);
  });
  it("component should render correct value", () => {
    expect(wrapper.find(".detailLocation").text()).toEqual(
      "123 Main Street, Tacoma, WA 98409"
    );
  });
  it("should render the component to match the snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
