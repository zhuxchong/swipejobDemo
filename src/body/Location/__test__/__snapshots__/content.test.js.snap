// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Description component should render the component to match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Location
    location="123 Main Street, Tacoma, WA 98409"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <h4>
          Location
        </h4>,
        <p
          className="detailLocation"
        >
          123 Main Street, Tacoma, WA 98409
        </p>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Location",
        },
        "ref": null,
        "rendered": "Location",
        "type": "h4",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "123 Main Street, Tacoma, WA 98409",
          "className": "detailLocation",
        },
        "ref": null,
        "rendered": "123 Main Street, Tacoma, WA 98409",
        "type": "p",
      },
    ],
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": true,
        "rules": Array [
          "
  border-top: solid gray 1px;
  border-bottom: solid gray 1px;
  width: 100%;
  padding: 3% 0%;
  display: flex;
  flex-direction: column;
  flex: 1.5;
  h4 {
    margin-top: 0;
    margin-bottom: 10px;
    text-align: center;
    line-height: 20px;
  }
  p {
    margin: 0;
    line-height: 20px;
    text-align: center;
    color: #0066cc;
    text-decoration: underline;
  }
  @media (max-width: 375px) {
    padding: 2% 0%;
    h4 {
      font-size: 15px;
      line-height: 12px;
    }
    p {
      font-size: 10px;
      line-height: 12px;
    }
  }
",
        ],
      },
      "displayName": "styled.div",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <h4>
            Location
          </h4>,
          <p
            className="detailLocation"
          >
            123 Main Street, Tacoma, WA 98409
          </p>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Location",
          },
          "ref": null,
          "rendered": "Location",
          "type": "h4",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "123 Main Street, Tacoma, WA 98409",
            "className": "detailLocation",
          },
          "ref": null,
          "rendered": "123 Main Street, Tacoma, WA 98409",
          "type": "p",
        },
      ],
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bdVaJa",
          "isStatic": true,
          "rules": Array [
            "
  border-top: solid gray 1px;
  border-bottom: solid gray 1px;
  width: 100%;
  padding: 3% 0%;
  display: flex;
  flex-direction: column;
  flex: 1.5;
  h4 {
    margin-top: 0;
    margin-bottom: 10px;
    text-align: center;
    line-height: 20px;
  }
  p {
    margin: 0;
    line-height: 20px;
    text-align: center;
    color: #0066cc;
    text-decoration: underline;
  }
  @media (max-width: 375px) {
    padding: 2% 0%;
    h4 {
      font-size: 15px;
      line-height: 12px;
    }
    p {
      font-size: 10px;
      line-height: 12px;
    }
  }
",
          ],
        },
        "displayName": "styled.div",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-bdVaJa",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
